src/core/Global.ttslua

```diff
@@ -825,3 +779,146 @@ function urldecode(str)
     function (h) return string.char(tonumber(h, 16)) end)
   return str
 end
+
+---------------------------------------------------------
+-- Option Panel related functionality
+---------------------------------------------------------
+
+-- loads the default options
+function onClick_defaultSettings()
+  print("Dummy: Load default settings")
+end
+
+-- called by toggling an option
+function onClick_toggleOption(_, id)
+  local state = self.UI.getAttribute(id, "isOn")
+
+  -- flip state (and handle stupid "False" value)
+  if state == "False" then
+    state = true
+  else
+    state = false
+  end
+
+  self.UI.setAttribute(id, "isOn", state)
+  optionPanel[id] = state
+  applyOptionPanelChange(id, state)
+end
+
+-- sets the option panel to the correct state (corresponding to 'optionPanel')
+function updateOptionPanelState()
+  for id, enabled in pairs(optionPanel) do
+    if enabled then
+      self.UI.setAttribute(id, "isOn", true)
+    end
+  end
+end
+
+-- handles the applying of option selections and calls the respective functions based
+---@param id String ID of the option that was selected or deselected
+---@param state Boolean State of the option (true = enabled)
+function applyOptionPanelChange(id, state)
+  -- option: Snap tags
+  if id == "useSnapTags" then
+    playmatAPI.setLimitSnapsByType(state, "All")
+
+  -- option: Draw 1 button
+  elseif id == "showDrawButton" then
+    playmatAPI.showDrawButton(state, "All")
+
+  -- option: Clickable clue counters
+  elseif id == "useClueClickers" then
+    playmatAPI.clickableClues(state, "All")
+
+    -- update master clue counter
+    getObjectFromGUID("4a3aa4").setVar("useClickableCounters", state)
+  
+  -- option: Show token arranger
+  elseif id == "showTokenArranger" then
+    -- delete previously pulled out tokens
+    for _, token in ipairs(getObjectsWithTag("to_be_deleted")) do token.destruct() end
+
+    spawnOrRemoveHelper(state, "Token Arranger", {-42.3, 1.4, -46.5})
+
+  -- option: Show clean up helper
+  elseif id == "showCleanUpHelper" then
+    spawnOrRemoveHelper(state, "Clean Up Helper", {-68, 1.6, 35.5})
+  
+  -- option: Show hand helper for each player
+  elseif id == "showHandHelper" then
+    spawnOrRemoveHelper(state, "Hand Helper", {-50.84, 1.6, 7.02}, {0, 270, 0}, "White")
```

Comment:
There's a lot of specific positioning here.  Can we find existent objects and derive the position relative to that?  Method could take a GUID and a relative position (like "above") and calculate a meaningful position.  Hooking to other objects would mean not needing to update if we rearranged things (though deletion would break it, so... up to you)

Reply:
Good idea! Will change this once we have a more "final" table layout.